src/djcytoscape/tests/test_views.py

```diff
@@ -81,11 +83,62 @@ def test_all_page_status_codes_for_teachers(self):
         self.assert200('djcytoscape:update', args=[self.map.id])
         self.assert200('djcytoscape:delete', args=[self.map.id])
 
+        self.assert200('djcytoscape:generate_unseeded')
+        self.assert200('djcytoscape:generate_map', kwargs={'quest_id': 1, 'scape_id': self.map.id})
+
         # These will need their own tests:
         # self.assert200('djcytoscape:regenerate', args=[self.map.id])
         # self.assert200('djcytoscape:regenerate_all')
-        # self.assert200('djcytoscape:generate_map', kwargs={'quest_id': 1, 'scape_id': 1})
-        # self.assert200('djcytoscape:generate_unseeded')
+    
+    def test_ScapeGenerateMap__POST(self):
+        """ Assert a teacher can generate a map using ScapeGenerateMapView """
+        self.client.force_login(self.test_teacher)
+
+        # generate form datas
+        content_type = ContentType.objects.filter(CytoScape.ALLOWED_INITIAL_CONTENT_TYPES).first()
+        object_ = content_type.model_class().objects.first()
+
+        form_data = generate_form_data(model_form=GenerateQuestMapForm, name='New Name')
+        form_data.update({'initial_content_object': f'{content_type.id}-{object_.id}'}) 
+
+        # response tests
+        response = self.client.post(reverse('djcytoscape:generate_unseeded'), data=form_data)
+        self.assertEqual(response.status_code, 302)
```

Comment:
Can you verify in the test that it is redirecting where expected to?

Reply:
done